ethexe/service/src/lib.rs

```diff
@@ -535,340 +453,278 @@ impl Service {
+                                    s.receive_block_commitments(aggregated_block_commitments)
+                                })
+                            })?;
+                        }
+                        ObserverEvent::Blob { code_id, code } => {
+                            // TODO: spawn blocking here?
+                            let valid = processor.process_upload_code_raw(code_id, code.as_slice())?;
+
+                            let code_commitments = vec![CodeCommitment { id: code_id, valid }];
+
+                            option_call(&mut validator, |v| {
+                                let aggregated_code_commitments = v.aggregate(code_commitments)?;
+
+                                option_call(&mut network, |n| {
+                                    log::debug!("Publishing code commitments to network...");
+                                    n.publish_message(
+                                        NetworkMessage::PublishCommitments {
+                                            codes: Some(aggregated_code_commitments.clone()),
+                                            blocks: None,
+                                        }
+                                        .encode(),
+                                    );
+                                    Ok(())
+                                })?;
+
+                                option_call(&mut sequencer, |s| {
+                                    log::debug!(
+                                        "Received ({}) signed code commitments from local validator...",
+                                        aggregated_code_commitments.len()
+                                    );
+                                    s.receive_code_commitments(aggregated_code_commitments)
+                                })
+                            })?;
+                        }
+                    }
+                },
+                Some(event) = sequencer.maybe_next() => {
+                    let Some(s) = sequencer.as_mut() else {
+                        unreachable!("couldn't produce event without sequencer");
+                    };
 
-                    validation_round_timer.stop();
-                }
-                Some(event) = maybe_await(network_receiver.as_mut().map(|rx| rx.recv())) => {
                     match event {
-                        NetworkReceiverEvent::Message { source, data } => {
-                            log::debug!("Received a network message from peer {source:?}");
-
-                            let result = Self::process_network_message(
-                                data.as_slice(),
-                                &db,
-                                validator.as_mut(),
-                                sequencer.as_mut(),
-                                network_sender.as_mut(),
-                            );
+                        SequencerEvent::CollectionRoundEnded { block_hash: _} => {
```

Comment:
`cargo fmt` does not work for macros
```suggestion
                        SequencerEvent::CollectionRoundEnded { block_hash: _ } => {
```

Reply:
https://github.com/tokio-rs/tokio/issues/5441#issuecomment-1427088486